service/storage_service.cc

```diff
@@ -397,6 +402,21 @@ void storage_service::prepare_to_join(std::vector<inet_address> loaded_endpoints
         }
     }
 
+    if (replacing_a_node_with_same_ip) {
+        slogger.info("Replacing a node with same IP address, my address={}, node being replaced={}",
+                    get_broadcast_address(), get_broadcast_address());
+        slogger.info("Update tokens for replacing node early, replacing node has the same IP address of the node being replaced");
+        _token_metadata.update_normal_tokens(_bootstrap_tokens, get_broadcast_address());
```

Comment:
We discussed this in the https://github.com/scylladb/scylla/pull/5154. The idea is to make the replacing node has the same  view of the token ring as the rest of the cluster.

Reply:
Yes, but it's not clear to the view would not be the same without this change.

Reply:
because the replacing node adds the tokens before the replacing operation is done while the other nodes adds the tokens after the replacing operation is done.